core_types/src/agent_id.rs

```diff
@@ -0,0 +1,156 @@
+//! Agent Ids are 64 bytes long.
+//! - The first 32 bytes 0-32 are the public side of a libsodium signing keypair
+//! - The next 32 bytes 32-64 are the public side of a libsodium kx keypair
+//! When displayed to users 2 reed-solomon parity bytes are appended to the end,
+//! and the resulting buffer is base64url encoded.
+//! (https://tools.ietf.org/html/rfc4648#section-5)
+
+use std::convert::TryFrom;
+
+use super::error::HolochainError;
+use reed_solomon::{Decoder, Encoder};
+
+const PARITY_LEN: usize = 2;
+
+/// Represents a public agent identity
+pub struct AgentIdPub([u8; 64]);
+
+impl AgentIdPub {
+    /// get the signing public key portion of this identity
+    pub fn get_sig(&self) -> &[u8; 32] {
+        array_ref![self.0, 0, 32]
+    }
+
+    /// get the encryption (kx) public key portion of this identity
+    pub fn get_enc(&self) -> &[u8; 32] {
+        array_ref![self.0, 32, 32]
+    }
+}
+
+impl std::fmt::Display for AgentIdPub {
```

Comment:
Why are you implementing this by hand. You could just `#[derive(Debug)]` on the struct

Reply:
You'd think so, right? Apparently rust can only derive Debug for u8 arrays up to 32 bytes... If you know a way of doing this, I'm all ears.

Reply:
🙈 Oh boy... this again. Well... I guess we can't do any better than this